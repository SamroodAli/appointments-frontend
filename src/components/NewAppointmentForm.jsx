import {
  useMutation,
  useQueryClient,
} from 'react-query';
import PropTypes from 'prop-types';
import postAppointments from '../api/postAppointments';

const Form = ({ id, name }) => {
  const queryClient = useQueryClient();
  const mutation = useMutation(postAppointments, {
    onSuccess: () => {
      // Invalidate and refetch
      queryClient.invalidateQueries('appointments');
    },
  });

  const onSubmit = (e) => {
    e.preventDefault();
    mutation.mutate(id);
  };

  return (
    <form onSubmit={onSubmit}>
      <label htmlFor="name">
        <input value={name} />
      </label>
      <input type="submit" value="Submit" />
    </form>
  );
};

Form.propTypes = {
  id: PropTypes.number.isRequired,
  name: PropTypes.string.isRequired,
};

export default Form;
