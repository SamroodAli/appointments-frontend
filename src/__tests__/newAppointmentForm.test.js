import renderer from 'react-test-renderer';
// import { render } from '@testing-library/react';
import NewAppointmentForm from '../components/NewAppointmentForm';
import Wrapper from '../testWrapper';

test('NewAppointmentForm snapshot testing', () => {
  const tree = renderer.create(
    <Wrapper>
      <NewAppointmentForm id="1" name="Samrood" color="indigo" />
    </Wrapper>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
