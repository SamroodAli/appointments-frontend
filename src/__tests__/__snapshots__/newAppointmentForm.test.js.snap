// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewAppointmentForm snapshot testing 1`] = `
<div
  className="flex my-12 justify-center xl:h-full items-center bg-indigo-300 mx-auto w-8/12 ring-2 ring-indigo-900"
>
  <form
    className="text-indigo-900"
    onSubmit={[Function]}
  >
    <div
      className="my-2 "
    >
      <label
        htmlFor="teacher"
      >
        Teacher
        <input
          className="px-2 w-full"
          id="teacher"
          readOnly={true}
          value="Samrood"
        />
      </label>
    </div>
    <div
      className="my-2"
    >
      <label
        htmlFor="username"
      >
        Student
        <input
          className="px-2 w-full"
          id="username"
          readOnly={true}
          value=""
        />
      </label>
    </div>
    <div
      className="my-2"
    >
      <label
        htmlFor="date"
      >
         
        Choose Date
        <div
          className="react-datepicker-wrapper"
        >
          <div
            className="react-datepicker__input-container"
          >
            <input
              className=""
              disabled={false}
              id="date"
              onBlur={[Function]}
              onChange={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              readOnly={false}
              type="text"
              value="12/14/2021"
            />
          </div>
        </div>
      </label>
      <div
        className="my-2"
      >
        <label
          htmlFor="time"
        >
          Choose Time
          <select
            className="px-2 w-full"
            onChange={[Function]}
            value="UTC +1 9:00"
          >
            <option
              value="UTC +1 9:00"
            >
              UTC +1 9:00
            </option>
            <option
              value="UTC +1 12:00"
            >
              UTC +1 12:00
            </option>
            <option
              value="UTC +1 15:00"
            >
              UTC +1 15:00
            </option>
          </select>
        </label>
      </div>
      <input
        className="animate-pulse bg-indigo-600 hover:bg-{color }-900 p-2 mt-3 w-full text-white focus:outline-none focus:ring-2 focus:ring-indigo-600 focus:ring-opacity-50"
        type="submit"
        value="Book an appointment"
      />
    </div>
  </form>
</div>
`;
